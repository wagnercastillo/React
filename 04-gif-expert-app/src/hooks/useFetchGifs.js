import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (categoria) => {
	const [state, setState] = useState({
		data: [],
		loading: true,
	});

	useEffect(() => {
		getGifs(categoria).then((img) => {
			setState({
				data: img,
				loading: false,
			});
		});
	}, [categoria]);
	return state;
};
