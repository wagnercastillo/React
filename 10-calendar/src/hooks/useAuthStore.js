import { useDispatch, useSelector } from "react-redux"
import { calendarApi }  from "../api";
import { clearErrorMessage, onChecking, onLogin, onLogout } from "../store";

export const useAuthStore = () => {

    const { status, user, errorMessage } = useSelector( state => state.auth )
    const dispatch = useDispatch();

    const startLogin = async({email, password}) => {
        
        dispatch(onChecking());

        try {
            
            const { data } = await calendarApi.post('/auth', { email, password })

            localStorage.setItem('token', data.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            
            dispatch( onLogin({ name: data.name, uid: data.uid }) )
            


        } catch (error) {

            dispatch(onLogout('Credenciales Incorrectos'))
            
            setTimeout(() => {
                dispatch( clearErrorMessage() );
            }, 10);
        }
        
    }
    
    const startRegister = async({name, email, password}) => {

        dispatch(onChecking());
        try {

            const { data } = await calendarApi.post('/auth/new', { name, email, password })     

            localStorage.setItem('token', data.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            
            dispatch( onLogin({ name: data.name, uid: data.uid }) )


        } catch (error) {
            console.log(error);
            const errors = (
                error.response.data.errors?.password.msg || 
                error.response.data.errors?.name.msg || 
                error.response.data.errors?.email.msg ||
                error.response.data?.msg || '')

            dispatch(onLogout(errors))
            
            setTimeout(() => {
                dispatch( clearErrorMessage() );
            }, 10);
        }   

    }

    const checkAuthToken = async () => {

        const token = localStorage.getItem('token');
        if( !token ) return dispatch( onLogout() )

        try {

            const { data } = await calendarApi.get('/auth/renew');
            localStorage.setItem('token', data.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            dispatch( onLogin({ name: data.name, uid: data.uid }) )
            
        } catch (error) {
            localStorage.clear();
            dispatch( onLogout() )
        }
    }


    const startLogout = () => {
        localStorage.clear();
        dispatch(onLogout())
    }

    return {
        //* Propiedades
        errorMessage,
        status,
        user, 

        //* Metodos
        checkAuthToken,
        startLogin,
        startLogout,
        startRegister


    }
    
}

