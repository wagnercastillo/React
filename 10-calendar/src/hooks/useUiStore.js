import { useDispatch, useSelector } from "react-redux";
import { onOpenDateModal, onCloseDateModal } from '../store'

export const useUiStore = () => {

    const dispatch = useDispatch();

    const {
        isDateModalOpen
    } = useSelector(state => state.ui);

    const openDateModal = () => {
        dispatch(onOpenDateModal())
    }

    const closeDateModal = () => {
        dispatch(onCloseDateModal())
    }

    const toggleDateModal = () => {
        (isDateModalOpen)
            ? openDateModal()
            : closeDateModal();
    }

    return {
        // * Propiedades
        isDateModalOpen,

        // * Metodos
        openDateModal,
        closeDateModal,
        toggleDateModal,
    }

}