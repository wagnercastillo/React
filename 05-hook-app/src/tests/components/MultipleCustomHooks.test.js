import React from "react";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../src/hooks/useFetch";
// import { useCounter } from "../../../hooks/useCounter";
jest.mock("../../../src/hooks/useFetch");

describe("Pruebas en <MultipleCustomHooks />", () => {
	// useCounter.mockReturnValue({
	// 	counter: 10,
	// 	increment: () => {},
	// });

	test("debe mostrarse correctamente", () => {
		useFetch.mockReturnValue({
			data: null,
			loading: true,
			error: null,
		});

		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper).toMatchSnapshot();
	});

	test("debe de monstrar la informaciÃ³n", () => {
		useFetch.mockReturnValue({
			data: [{ author: "wagner", qoute: "hola" }],
			loading: true,
			error: null,
		});
		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper.find(".alert").exists()).toBe(true);
		// expect(wrapper.find(".mb-0").text().trim()).toBe("hola");
	});
});
