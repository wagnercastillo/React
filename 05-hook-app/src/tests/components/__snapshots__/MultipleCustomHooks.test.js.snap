// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Multiple Custom Hooks
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Cuote
  </button>
</div>
`;
